### ISSUE TITLE
Add ability to exclude sentences by specific owners in advanced search

### ISSUE THREAD

**Issue by alanfgh** [Tatoeba/tatoeba2#2008]
There are various scenarios in which I find myself wanting to exclude sentences by specific owners when I perform an advanced search. The most common, and probably the most generalizable to other people, is when I'm doing corpus maintenance, specifically looking through recently-written English sentences for errors. If I exclude the sentences by prolific owners whose sentences are mostly good, it becomes much more efficient to find the bad ones. I can then further increase my efficiency by narrowing in on the owners of bad sentences and looking through all of their sentences. However, there's no easy way to do this given the current advanced search.

**Comment by ckjpn** [comment]
This would be useful, especially if it could be done like tags and allowed a comma-delimited list of usernames.

Additionally, it would be useful to be able to list more than one username in the current "Owner:" field.

**Comment by ckjpn** [comment]
Maybe instead of an additional field in the search, it would be possible to use the same field and just put a minus in front of a username or usernames like the following

-CK, -Hybrid, -AlanF_US

**Comment by ckjpn** [comment]
If the programming is similar, perhaps at the same time we should consider adding the following options.

Tags:  -imperative
(Allow searching for sentences not yet tagged with a certain tag.)

Allow searching sentences limited to those who are not self-identified native speakers.
This would perhaps also help those find sentences that might more likely need corrections.

Allow limiting results to sentences that are not yet on a certain list.

**Comment by Inego** [comment]
What about the ability to create personal lists of users (like personal lists of sentences), and filtering by the list, and not by explicitly enumerated users?

**Comment by jiru** [comment]
Yes, members of Tatoeba certainly want to have their own "view" of the corpus. CK is doing it with his giant list of proofread English sentences. Trang started to implement the Collection feature with a similar goal in mind as far as I know.

I think this idea of "views" is quite of a complex matter, so we need to better understand the needs of our members if we want to implement a good solution that works for everyone.

Yanis, could you elaborate on your use case? In what context are you using Tatoeba? What kind of sentences are you looking for? What are you trying to achieve when filtering by a list of users? What kind of user list do you have in mind? How would you decide who's on a given list of yours?

**Comment by jiru** [comment]
Well, forget about my last comment. I wrote it by email as a reply to Yanis’ last comment, thinking that he opened a new issue.

**Comment by jiru** [comment]
@alanfgh Isn’t there a way to exclude users without having to explicitly list them? For example, exclude users who
- have more than 100 sentences
- registered more than a month ago
- are ranked as advanced contributor or higher

What about your other scenarios?

**Comment by sacredceltic** [comment]
Le déploiement de tous ces efforts pour l’atteinte de ces admirables buts semble tellement essentiel ^^

**Comment by alanfgh** [comment]
@jiru, another scenario in which I might want to exclude sentences by one or more users would be a regular search. Let's say I find that a particular contributor's style doesn't give me what they're looking for. Maybe their sentences are too slangy, or too long, or unnatural. I haven't really run into that scenario personally, but I'm thinking of a prolific contributor in Turkish whose sentences are often criticized. I can imagine that people would want the ability to exclude his sentences from the search.

As for your proposed alternative criteria, they involve parameters ("more than 100", "more than a month ago") rather than simple usernames and thus require more work (on the part of both the software developers and the users). More importantly, they don't fit any real-world needs of mine, whereas I have run into situations where being able to exclude one or more users would be helpful.

**Comment by sacredceltic** [comment]
You should change the issue title to : Set up apartheid on Tatoeba ;-)

**Comment by Inego** [comment]
@sacredceltic These are all instruments to setup your personal search filtering. I wouldn't use the word "apartheid" here.
Also, suppose I have some friends here on Tatoeba or a list of contributors whose recent sentences I'd like to read (and translate) from time to time? It'd be great to have my own list of these people to quickly filter by it.

**Comment by ckjpn** [comment]
>  ... list of contributors whose recent sentences I'd like to read (and translate) from time to time

This is a good point.  Perhaps it would not only be useful to exclude certain members' sentences from searches for the reason stated above, but also to only include certain members' sentences, excluding all others from searches.

Maybe the existing field could include the possibility to search this way.

- CK, Hybrid, AlanF_US
-- Only show sentences by these 3 members, excluding sentences by all other members

- -CK, -Hybrid, -AlanF_US
-- Don't show sentences by these 3 members, but show sentences by other members.

I basically suggested bother of these above, and just included this comment for clarification.
https://github.com/Tatoeba/tatoeba2/issues/2008#issuecomment-553692316

**Comment by Inego** [comment]
I think @ckjpn's suggestion can be improved a bit. Putting minus sign in front of each name is redundant (because, really, you can't include some names and exclude some other names at the same time). Either the whole list is to limit to it, or to exclude. So, basically, the following combination of new features will do the work:
* Support specifying multiple, comma separated names of contributors
* "Exclude / limit to contributors" dropdown to toggle the filtering behavior (like the same dropdown in the Translations block).
This doesn't seem to be difficult to implement.

**Comment by ckjpn** [comment]
Good point. :)
